include/pgduckdb/pgduckdb_types.hpp

```diff
@@ -1,25 +1,20 @@
 #pragma once
 
-#include "duckdb.hpp"
-#include "duckdb/common/shared_ptr.hpp"
-#include "duckdb/common/extra_type_info.hpp"
-
-extern "C" {
-#include "postgres.h"
-#include "executor/tuptable.h"
-}
-
-#include "pgduckdb/scan/postgres_scan.hpp"
+#include "pgduckdb/pg/declarations.hpp"
 
 namespace pgduckdb {
 
+class PostgresScanGlobalState;
+class PostgresScanLocalState;
+typedef uint64_t idx_t; // From duckdb.h
```

Comment:
Why do we need to declare these here. Especially the `idx_t` seems strange to declare. Why not include `duckdb.hpp` anymore?

I guess PostgresScanGlobalState and PostgresScanLocalState forward declarations are useful because `include/pgduckdb/scan/postgres_scan.hpp` is including `postgres.h`. Can we change that instead?

Reply:
Because `pgduckdb_types.hpp` is included in a bunch of file I wanted to remain very conservative with adding headers, especially `duckdb.h`. Instead of declaring `idx_t` I've updated the interface to remain consistent, that's even better I think.

> because include/pgduckdb/scan/postgres_scan.hpp is including postgres.h. Can we change that instead?

yep, it's coming :-)